src/test/java/skeleton/RestSteps.java

```diff
@@ -60,12 +62,53 @@ public void i_add_users(List<User> users) {
         }
     }
 
+    @When("I delete user (.*?)$")
+    public void i_delete_user(String email) {
+        doRequest(
+                Unirest.delete(usersBaseUrl).queryString("email", email)
+        );
+
+    }
+
+    @When("I delete all users")
+    public void i_delete_all_users() {
+        doRequest(Unirest.delete(usersBaseUrl));
+    }
+
+    @When("I edit user (.*?) with (.*?) and (.*?)$")
+    public void i_edit_user(String email, String newFirstname, String newLastname) {
+
+        doRequest(
+                Unirest.post(usersBaseUrl + "/edit")
+                        .queryString("email", email)
+                        .body("{\"firstName\":\"" + newFirstname + "\",\"lastName\":\"" + newLastname + "\"}")
+                        .getHttpRequest()
+        );
+
+    }
+
     @Then("^there are (\\d+) users$")
     public void there_are_users(int expectedUsersCount) {
         doRequest(Unirest.get(usersBaseUrl));
         int actualUsersCount = response.getBody().getObject().getJSONArray("users").length();
         assertEquals("BBB!", expectedUsersCount, actualUsersCount);
     }
+
+    @Then("User (.*?) has firstname (.*?) and lastname (.*?)$")
+    public void user_has_credentials(String email, String firstname, String lastname) {
+        doRequest(Unirest.get(usersBaseUrl));
+        JSONArray users = response.getBody().getObject().getJSONArray("users");
+        int index;
+        for(index = 0; index < users.length(); index++) {
+            JSONObject userRecord = users.getJSONObject(index);
+            if (userRecord.getString("email").equals(email)) break;
+
+        }
+        String actualFirstname = users.getJSONObject(index).getString("first_name");
+        String actualLastname = users.getJSONObject(index).getString("last_name");
+
+        assertTrue(actualFirstname.equals(firstname) && actualLastname.equals(lastname));
```

Comment:
Such assertTrue is not good practice. It's not clear what has really happened.
Better solution 1 is to have two assertions:
assertEquals(firstname, actualFirstName);
assertEquals(lastname, actualLastName);

Better than better solution 1 is better solution 2. It requires some additional work but I'll always advocate for using such an approach.
There is object User. Method equals' (and therefore 'hashcode') might be updated. 1 more constructor with parameter JSONObject should be added. So code will look something like:
User expectedUser = new User(email, expectedFirstName, expectedLastName);
Map<String, User> userMap = new HashMap<String, User>();
for(index = 0; index < users.length(); index++) {
  JSONObject userRecord = users.getJSONObject(index);
  userMap.put(userRecord.getString("email"), userRecord);
}
User actualUser = userList.get(email);
assertEquals("User has wrong credentials", expectedUser, actualUser);